File: core/src/nl/t64/game/rpg/screens/ScreenToLoad.java
package nl.t64.game.rpg.screens;

import com.badlogic.gdx.Screen;
import com.badlogic.gdx.scenes.scene2d.Stage;
import com.badlogic.gdx.scenes.scene2d.actions.Actions;
import com.badlogic.gdx.scenes.scene2d.ui.Image;
import nl.t64.game.rpg.Utils;
import nl.t64.game.rpg.constants.Constant;
import nl.t64.game.rpg.constants.ScreenType;


public abstract class ScreenToLoad implements Screen {

    protected final Stage stage;

    protected ScreenToLoad() {
        this.stage = new Stage();
    }

    public ScreenUI getScreenUI() {
        throw new IllegalCallerException("ScreenUI not implemented here.");
    }

    @Override
    public void resize(int width, int height) {
        // empty
    }

    @Override
    public void pause() {
        // empty
    }

    @Override
    public void resume() {
        // empty
    }

    protected void fadeParchment() {
        var screenshot = (Image) stage.getActors().get(0);
        var parchment = (Image) stage.getActors().get(1);
        stage.clear();
        setBackground(screenshot, parchment);
        parchment.addAction(Actions.sequence(Actions.fadeOut(Constant.FADE_DURATION),
                                             Actions.run(() -> Utils.getScreenManager().setScreen(ScreenType.WORLD))));
    }

    protected void setBackground(Image screenshot, Image parchment) {
        stage.addActor(screenshot);
        stage.addActor(parchment);
    }

}

File: core/src/nl/t64/game/rpg/screens/ScreenToLoad.kt
package nl.t64.game.rpg.screens

import com.badlogic.gdx.Screen
import com.badlogic.gdx.scenes.scene2d.Stage
import com.badlogic.gdx.scenes.scene2d.actions.Actions
import com.badlogic.gdx.scenes.scene2d.ui.Image
import nl.t64.game.rpg.Utils.screenManager
import nl.t64.game.rpg.constants.Constant
import nl.t64.game.rpg.constants.ScreenType


abstract class ScreenToLoad : Screen {

    protected val stage: Stage = Stage()

    open fun getScreenUI(): ScreenUI {
        throw IllegalStateException("ScreenUI not implemented here.")
    }

    override fun resize(width: Int, height: Int) {
        // empty
    }

    override fun pause() {
        // empty
    }

    override fun resume() {
        // empty
    }

    protected fun fadeParchment() {
        val screenshot = stage.actors[0] as Image
        val parchment = stage.actors[1] as Image
        stage.clear()
        setBackground(screenshot, parchment)
        parchment.addAction(Actions.sequence(Actions.fadeOut(Constant.FADE_DURATION),
                                             Actions.run { screenManager.setScreen(ScreenType.WORLD) }))
    }

    fun setBackground(screenshot: Image, parchment: Image) {
        stage.addActor(screenshot)
        stage.addActor(parchment)
    }

}
